applications/GeoMechanicsApplication/custom_conditions/U_Pw_face_load_condition.cpp

```diff
@@ -29,46 +29,42 @@ Condition::Pointer UPwFaceLoadCondition<TDim, TNumNodes>::Create(IndexType
-    const GeometryType::IntegrationPointsArrayType& IntegrationPoints =
-        Geom.IntegrationPoints(this->GetIntegrationMethod());
-    const unsigned int NumGPoints = IntegrationPoints.size();
-    const unsigned int LocalDim   = Geom.LocalSpaceDimension();
+    const GeometryType&                             r_geom = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& r_integration_points =
+        r_geom.IntegrationPoints(this->GetIntegrationMethod());
+    const unsigned int NumGPoints = r_integration_points.size();
 
     // Containers of variables at all integration points
-    const Matrix& NContainer = Geom.ShapeFunctionsValues(this->GetIntegrationMethod());
-    GeometryType::JacobiansType JContainer(NumGPoints);
-    for (unsigned int i = 0; i < NumGPoints; ++i)
-        (JContainer[i]).resize(TDim, LocalDim, false);
-    Geom.Jacobian(JContainer, this->GetIntegrationMethod());
+    const Matrix& r_n_container = r_geom.ShapeFunctionsValues(this->GetIntegrationMethod());
+    GeometryType::JacobiansType j_container(NumGPoints);
+    for (auto& j : j_container)
+        j.resize(TDim, r_geom.LocalSpaceDimension(), false);
+    r_geom.Jacobian(j_container, this->GetIntegrationMethod());
 
     // Condition variables
-    array_1d<double, TNumNodes * TDim> FaceLoadVector;
-    ConditionUtilities::GetFaceLoadVector<TNumNodes>(FaceLoadVector, Geom);
-    BoundedMatrix<double, TDim, TNumNodes * TDim> Nu = ZeroMatrix(TDim, TNumNodes * TDim);
-    array_1d<double, TDim>                        TractionVector;
-    array_1d<double, TNumNodes * TDim>            UVector;
+    array_1d<double, TNumNodes * TDim> face_load_vector;
+    ConditionUtilities::GetFaceLoadVector<TDim, TNumNodes>(face_load_vector, r_geom);
 
-    // Loop over integration points
-    for (unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+    for (unsigned int integration_point = 0; integration_point < NumGPoints; ++integration_point) {
         // Compute traction vector
+        array_1d<double, TDim> traction_vector;
         ConditionUtilities::InterpolateVariableWithComponents<TDim, TNumNodes>(
-            TractionVector, NContainer, FaceLoadVector, GPoint);
+            traction_vector, r_n_container, face_load_vector, integration_point);
 
         // Compute Nu Matrix
-        ConditionUtilities::CalculateNuMatrix<TDim, TNumNodes>(Nu, NContainer, GPoint);
+        BoundedMatrix<double, TDim, TNumNodes * TDim> nu = ZeroMatrix(TDim, TNumNodes * TDim);
+        ConditionUtilities::CalculateNuMatrix<TDim, TNumNodes>(nu, r_n_container, integration_point);
 
         // Compute weighting coefficient for integration
-        double integration_coefficient = ConditionUtilities::CalculateIntegrationCoefficient<TDim, TNumNodes>(
-            JContainer[GPoint], IntegrationPoints[GPoint].Weight());
+        auto integration_coefficient = ConditionUtilities::CalculateIntegrationCoefficient(
```

Comment:
Let's make this one `const`:
```suggestion
        const auto integration_coefficient = ConditionUtilities::CalculateIntegrationCoefficient(
```

Reply:
done